server/engine/src/main/java/org/eclipse/lsp/cobol/core/visitor/CobolVisitor.java

```diff
@@ -156,6 +156,38 @@ private List<Node> makeFunctionReferenceNodes(FunctionNameContext fnCtx) {
         .orElse(ImmutableList.of());
   }
 
+  @Override
+  public List<Node> visitFunctionRepositoryClause(FunctionRepositoryClauseContext ctx) {
+    Optional<Locality> statementLocality = retrieveLocality(ctx, extendedDocument, copybooks);
+    if (statementLocality.isPresent()) {
+      boolean isIntrinsic = ctx.INTRINSIC() != null;
+      TerminalNode all = ctx.ALL();
+      if (Objects.nonNull(all)) {
+        Optional<Locality> allImplicitLocality = retrieveLocality(ctx.ALL(), extendedDocument, copybooks);
+        if (allImplicitLocality.isPresent()) {
+          return  ImmutableList.of(new FunctionDeclaration(allImplicitLocality.get()));
+        }
```

Comment:
This function cannot fallthrough into the other logic branch...
```suggestion
        return retrieveLocality(ctx.ALL(), extendedDocument, copybooks).map(FunctionDeclaration::new).get();
```

Reply:
I think I didn't understand . We may not want to fallthrough based on below dig
![image](https://github.com/user-attachments/assets/f31e12b4-3f96-4760-a7b4-c5a3901cfaf5)

Reply:
`cannot` should have been `must not` - if `retrieveLocality` returns empty option we will fallthrough.

Reply:
done